### PR TITLE
[feature/APPC-1117] - Successful payment bonus animation

### PR DESCRIPTION
**Pull request by joaopmgouveia**
**What does this PR do?**

   This PR implements the animation that's shown after completing a purchase with a payment method that is eligible for bonus.

**Database changed?**

   No

**Where should the reviewer start?**

- AdyenAuthorizationFragment.java
- OnChainBuyFragment.java
- fragment_iab_transaction_completed.xml

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APPC-1117](https://aptoide.atlassian.net/browse/APPC-1117)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1117](https://aptoide.atlassian.net/browse/APPC-1117)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/ui/iab/AdyenAuthorizationPresenter.java [hunk 1]
```diff
@@ -203,22 +203,23 @@ private void onViewCreatedSelectPaymentMethod() {
...
+            .flatMapCompletable(bundle -> Completable.fromAction(() -> {
               waitingResult = false;
               sendPaymentEvent();
               sendRevenueEvent();
-              navigator.popView(bundle);
+              view.showSuccess();
             })
-            .doOnSuccess(__ -> view.showSuccess()))
-        .subscribe(__ -> {
+                .andThen(Completable.timer(view.getAnimationDuration(), TimeUnit.MILLISECONDS))
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/AdyenAuthorizationPresenter.java:220]: I think it’s better to pass the duration on the present method (or in the presenter constructor) instead as we should avoid presenter methods asking for view states/attributes

File: app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyFragment.java [hunk 2]
```diff
@@ -3,8 +3,10 @@
 import android.content.Context;
 import android.content.pm.ApplicationInfo;
 import android.content.pm.PackageManager;
+import android.graphics.Typeface;
 import android.graphics.drawable.Drawable;
 import android.os.Bundle;
+import android.util.Log;
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyFragment.java]: Codestyle/Import.

File: app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyFragment.java [hunk 3]
```diff
@@ -50,6 +55,7 @@
 
   public static final String APP_PACKAGE = "app_package";
   public static final String TRANSACTION_BUILDER_KEY = "transaction_builder";
+  public static final String BONUS_KEY = "bonus";
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyFragment.java]: Can be private. Same for the other 2 (although they were already there)

File: app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyPresenter.java [hunk 4]
```diff
@@ -159,7 +159,7 @@ private Completable showPendingTransaction(Payment transaction) {
             .map(payment -> buildBundle(payment, transaction.getOrderReference()))
             .flatMapCompletable(bundle -> Completable.fromAction(view::showTransactionCompleted)
                 .subscribeOn(AndroidSchedulers.mainThread())
-                .andThen(Completable.timer(1, TimeUnit.SECONDS))
+                .andThen(Completable.timer(view.getAnimationDuration(), TimeUnit.MILLISECONDS))
```
Comment (JDSM01) [inline app/src/main/java/com/asfoundation/wallet/ui/iab/OnChainBuyPresenter.java:162]: same as above

**Review by JDSM01 — CHANGES_REQUESTED** [review]
When you’re on the complete screen and you rotate the screen, it’s suppose to keep that screen in landscape (unless it’s being killed from the previous activity, which it looks by the logs, but we can talk about it)

**Review by JDSM01 — COMMENTED** [review]
Ignore the general error reported before

**Review by carvalho2707 — APPROVED** [review]

**Pull request merged**